packages/indexer/src/utils/price.helpers.ts

```diff
@@ -20,6 +20,29 @@ export interface CoinGeckoResponse {
 	}
 }
 
+export interface CoinGeckoTokenListToken {
+	chainId: number
+	address: string
```

Comment:
Does coingecko api provide us the dex pair?

Reply:
Yes. but that is a different endpoint with a lot of data for a token. Below is an example for a token address: 

`https://api.coingecko.com/api/v3/coins/ethereum/contract/0x1f9840a85d5af5bf1d1762f925bdaddc4201f984`

Reply:
Sorry, the return actually mentions all the market's listed for a token address, like cex names and dex pairs without actual pair address: 
<img width="859" height="381" alt="image" src="https://github.com/user-attachments/assets/88795634-2c1a-4d50-a5bc-51ce9ebba225" />

Reply:
So we have to find the dex pair ourselves

Reply:
That's how they fetch dex pairs automatically on CoinGecko website

Reply:
@royvardhan any luck with this?

Reply:
Yes gecko terminal has the endpoint.

`/networks/{network}/tokens/{token_address}/pools`

But I don't think this pr is blocked by this. We are storing token lists in the db, and when user selects a token, the FE(or maybe sdk) fetches the pool address.

Reply:
Hmmnn, what do you think about just indexing pools directly. They have a pools endpoint; the pools data returns the token addresses in these pools. We can probably then transform the result into what we want by extracting the token address, pool address, and dex information